src/control/client/config.go

```diff
@@ -80,52 +81,87 @@ type Configuration struct {
 	Ext             External
 }
 
-// newDefaultConfiguration creates a new instance of configuration struct
+// newDefaultAgentConfiguration creates a new instance of configuration struct
 // populated with defaults.
-func newDefaultConfiguration(ext External) *Configuration {
+func newDefaultAgentConfiguration(ext External) *Configuration {
 	return &Configuration{
 		SystemName:      defaultSystemName,
 		AccessPoints:    []string{fmt.Sprintf("localhost:%d", defaultControlPort)},
 		ControlPort:     defaultControlPort,
 		HostList:        []string{fmt.Sprintf("localhost:%d", defaultControlPort)},
 		RuntimeDir:      defaultRuntimeDir,
 		LogFile:         defaultLogFile,
-		Path:            defaultConfigPath,
-		TransportConfig: security.DefaultClientTransportConfig(),
+		Path:            defaultAgentConfigPath,
+		TransportConfig: security.DefaultAgentTransportConfig(),
 		Ext:             ext,
 	}
 }
 
-// GetConfig loads a configuration file from the path given,
+// newDefaultAdminConfiguration creates a new instance of configuration struct
+// populated with defaults.
+func newDefaultAdminConfiguration(ext External) *Configuration {
+	return &Configuration{
+		SystemName:      defaultSystemName,
+		AccessPoints:    []string{fmt.Sprintf("localhost:%d", defaultControlPort)},
+		ControlPort:     defaultControlPort,
+		HostList:        []string{fmt.Sprintf("localhost:%d", defaultControlPort)},
+		RuntimeDir:      defaultRuntimeDir,
+		LogFile:         defaultLogFile,
+		Path:            defaultAdminConfigPath,
+		TransportConfig: security.DefaultAdminTransportConfig(),
+		Ext:             ext,
+	}
+}
+
+// GetAdminConfig loads a configuration file from the path given,
 // or from the default location if none is provided.  It returns a populated
 // Configuration struct based upon the default values and any config file overrides.
-func GetConfig(log logging.Logger, inPath string) (*Configuration, error) {
-	c := NewConfiguration()
+func GetAdminConfig(log logging.Logger, inPath string) (*Configuration, error) {
+	c := NewAdminConfiguration()
+	if err := getConfig(log, inPath, c); err != nil {
+		return nil, errors.Wrapf(err, "Unable to load Admin configuration")
+	}
+	return c, nil
+}
+
+// GetAgentConfig loads a configuration file from the path given,
+// or from the default location if none is provided.  It returns a populated
+// Configuration struct based upon the default values and any config file overrides.
+func GetAgentConfig(log logging.Logger, inPath string) (*Configuration, error) {
+	c := NewAgentConfiguration()
+	if err := getConfig(log, inPath, c); err != nil {
+		return nil, errors.Wrapf(err, "Unable to load Agent configuration")
+	}
+	return c, nil
+}
+
+func getConfig(log logging.Logger, inPath string, c *Configuration) error {
```

Comment:
IMO should be "updateConfigFromPath" or something like that, since it's updating an existing config rather than constructing a new one

Reply:
I'd imagine the entire config test suite is going to be changed when mike reworks the config code in master.